src/Sylius/Behat/Context/Ui/Shop/LocaleContext.php

```diff
@@ -15,54 +15,66 @@
 
 use Behat\Behat\Context\Context;
 use Sylius\Behat\Page\Shop\HomePageInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
 use Sylius\Component\Locale\Context\LocaleNotFoundException;
+use Sylius\Component\Locale\Model\LocaleInterface;
 use Webmozart\Assert\Assert;
 
 final class LocaleContext implements Context
 {
-    private HomePageInterface $homePage;
+    public function __construct(
+        private HomePageInterface $homePage,
+        private SharedStorageInterface $sharedStorage
+    ) {
+    }
 
-    public function __construct(HomePageInterface $homePage)
+    /**
+     * @Given I switched the shop's locale to :locale
+     * @Given I have switched to the :locale locale
+     * @When I switch to the :locale locale
+     * @When I change my locale to :locale
+     */
+    public function iSwitchTheLocaleToTheLocale(LocaleInterface $locale): void
     {
-        $this->homePage = $homePage;
+        $this->homePage->open();
+        $this->homePage->switchLocale($locale->getName('en_US'));
+
+        $this->sharedStorage->set('locale', $locale);
     }
 
     /**
-     * @Given I switched the shop's locale to :name
-     * @Given I have switched to the :name locale
-     * @When I switch to the :name locale
-     * @When I change my locale to :name
+     * @When I show homepage with the locale :locale
      */
-    public function iSwitchTheLocaleToTheLocale(string $name): void
+    public function iShowHomepageWithTheLocale(LocaleInterface $locale): void
     {
-        $this->homePage->open();
-        $this->homePage->switchLocale($name);
+        $this->homePage->tryToOpen(['_locale' => $locale->getCode()]);
```

Comment:
Why did you change that? I would be for leaving as it was
```suggestion
        $this->homePage-> switchLocale($locale->getName());
```

Reply:
I didn't, GitHub shows diff it in a strange way. This change you highlighted is for the `iShowHomepageWithTheLocale`.